### PR TITLE
In-memory substitute

### PR DESCRIPTION
**Pull request by flintinatux**
mock yeah

To simplify testing, this adds an in-memory substitute enabled with `{ mock: true }`.  More details in the new docs.  

Also added these two helpers after attending the Eventide workshop:

- `follow`
- `writeMessage.initial`

## to review:
- [ ] Wait for the green
- [ ] Fire up `yarn docs` to read about the new stuff

### CODE REVIEW

File: test/Entity.js [hunk 1]
```diff
@@ -236,4 +242,9 @@ describe('Entity', () => {
       expect(version).to.equal(0)
     })
   })
+}
+
+describe('Entity', () => {
+  describe('in memory', test(memory))
+  describe('in postgres', test(postgres))
```
Comment (mgreystone) [inline test/Entity.js:249]: ❗

**Review by mgreystone — APPROVED** [review]
The in memory mock is awesome.

I'm not following what "follow" does (hah).

**Comment by flintinatux** [comment]
@mgreystone, `follow` tracks message precedence/provenance in larger message workflows.  Some details here:
http://docs.eventide-project.org/user-guide/messages-and-message-data/metadata.html#message-workflows

The idea is that one message follows another, and we can track that.  It's a thing in Eventide, so I added it as a thing you can use in Hermes if you want to be compatible with that.  Happy to talk about it offline if you like.

**Comment by flintinatux** [comment]
This has had a couple weeks to simmer.  I'm gonna go ahead and ship it now.

**Pull request merged**